src/test/java/com/walmartlabs/x12/common/AbstractStandardX12ParserTest.java

```diff
@@ -0,0 +1,97 @@
+/**
+Copyright (c) 2018-present, Walmart, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+package com.walmartlabs.x12.common;
+
+import com.walmartlabs.x12.X12Segment;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class AbstractStandardX12ParserTest {
+
+    AbstractStandardX12Parser standardParser;
+
+    @Before
+    public void init() {
+        standardParser = new MockStandardParser();
+    }
+
+    @Test
+    public void test_parseInterchangeControlHeader() {
```

Comment:
Add test cases which does not have expected segments. I.E. not `ISA` and not `GS`